kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/kommunicate/KmAttachmentsController.java

```diff
@@ -0,0 +1,198 @@
+     * -4: format empty, -10: exception, 1: function end
+     */
+    public int processFile(Uri selectedFileUri, AlCustomizationSettings alCustomizationSettings, PrePostUIMethods prePostUIMethods) {
+        if (selectedFileUri != null) {
+            String fileName;
+            try {
+                long maxFileSize = alCustomizationSettings.getMaxAttachmentSizeAllowed() * 1024 * 1024;
+                Cursor returnCursor =
+                        context.getContentResolver().query(selectedFileUri, null, null, null, null);
+                if (returnCursor != null) {
+                    int sizeIndex = returnCursor.getColumnIndex(OpenableColumns.SIZE);
+                    returnCursor.moveToFirst();
+                    Long fileSize = returnCursor.getLong(sizeIndex);
+                    returnCursor.close();
+                    if (fileSize > maxFileSize) {
+                        Utils.printLog(context, TAG, context.getResources().getString(R.string.info_attachment_max_allowed_file_size));
+                        //Toast.makeText(this, R.string.info_attachment_max_allowed_file_size, Toast.LENGTH_LONG).show();
+                        return -1;
+                    }
+                }
+                String mimeType = FileUtils.getMimeTypeByContentUriOrOther(context, selectedFileUri);
+                if (TextUtils.isEmpty(mimeType)) {
+                    return -2;
+                }
+                if (!checkMimeType(mimeType, alCustomizationSettings)) {
+                    //Toast.makeText(this, R.string.info_file_attachment_mime_type_not_supported, Toast.LENGTH_LONG).show();
+                    return -3;
+                }
+                String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+                //because images are selected multiple and quickly (milliseconds better in such a situation)
+                timeStamp = timeStamp + "_" + System.currentTimeMillis();
+                fileName = FileUtils.getFileName(context, selectedFileUri);
+
+                String fileFormat = FileUtils.getFileFormat(fileName);
+                String fileNameToWrite;
+                if (TextUtils.isEmpty(fileFormat)) {
+                    String format = FileUtils.getFileFormat(FileUtils.getFile(context, selectedFileUri).getAbsolutePath());
+                    if (TextUtils.isEmpty(format)) {
+                        return -4;
+                    }
+                    fileNameToWrite = timeStamp + "." + format;
+                } else {
+                    fileNameToWrite = timeStamp + "." + fileFormat;
+                }
+                File mediaFile = FileClientService.getFilePath(fileNameToWrite, context.getApplicationContext(), mimeType);
+                new FileTaskAsync(mediaFile, selectedFileUri, context, prePostUIMethods).execute((Void) null);
+            } catch (Exception e) {
+                e.printStackTrace();
+                return -10;
```

Comment:
@shubhamtewari whats -10 for?

Reply:
ive mention the reason for the returned values in the comments above the function

Reply:
can these be replaced with constants

Reply:
yes, replaced